### PR TITLE
Add a test to make sure all modules in the codebase are importable

### PR DESCRIPTION
**Pull request by albanD**
As per title, running import on any of these files lead to a crash.
I'm very curious how the code in them is used!

### CODE REVIEW

File: test/test_public_bindings.py [hunk 1]
```diff
@@ -234,6 +235,45 @@ def test_no_new_bindings(self):
         msg = f"torch._C had bindings that are not present in the allowlist:\n{difference}"
         self.assertTrue(torch_C_bindings.issubset(torch_C_allowlist_superset), msg)
 
+    # AttributeError: module 'torch.distributed' has no attribute '_shard'
+    @unittest.skipIf(IS_WINDOWS or IS_JETSON, "Distributed Attribute Error")
+    def test_modules_can_be_imported(self):
+        failure = []
+        succcess = []
```
Comment (janeyx99) [inline test/test_public_bindings.py:242]:
```suggestion
        success = []
```

Comment (janeyx99) [reply test/test_public_bindings.py:242]: you don’t use this?

**Review by janeyx99 — APPROVED** [review]

File: test/test_public_bindings.py [hunk 2]
```diff
@@ -234,6 +235,136 @@ def test_no_new_bindings(self):
...
+            "torch.distributed.rpc.backend_registry"
+            "torch.distributed.rpc.constants"
+            "torch.distributed.rpc.internal"
+            "torch.distributed.rpc.options"
+            "torch.distributed.rpc.rref_proxy"
+        )
+
+        print("Modules which failed to import")
+        print("\n".join(map(str, failures)))
+
```
Comment (malfet) [inline test/test_public_bindings.py:354]: Please don't print anything while running the tests (or use`logging`)
```suggestion
```

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**